tests/e2e/solidity/suites.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// Implements ping tests, requires network-runner cluster.
+package solidity
+
+import (
+	"fmt"
+	"os/exec"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+
+	"github.com/onsi/gomega"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/tests/e2e/runner"
+	"github.com/ava-labs/subnet-evm/tests/e2e/utils"
+)
+
+var vmId ids.ID
+
+const vmName = "subnetevm"
+
+func runHardhatTests(test string) {
+	cmd := exec.Command("npx", "hardhat", "test", test, "--network", "subnet")
+	cmd.Dir = "./contract-examples"
+	out, err := cmd.Output()
+	fmt.Println(string(out))
+	gomega.Expect(err).Should(gomega.BeNil())
+}
+
+func startSubnet(genesisPath string) {
+	runner.StartNetwork(vmId, vmName, genesisPath, utils.GetPluginDir())
+	utils.UpdateHardhatConfig()
+}
+
+var _ = utils.DescribePrecompile("[TX Allow List]", func() {
```

Comment:
should we add deployer allow list and native minter precompile tests as well? 

Actually I was thinking to spin-up ANR only one time and enable all possible precompiles in the same genesis and run all hardhat tests at the same time. Wouldn't that actually much faster as we wouldn't need to spin up a network for each precompile test? Plus we would catch any edge-cases where all of them are activated at the same time etc.

Reply:
I wanted to focus this PR on setting up the framework more than writing actual tests. I think those should be a follow-up effort.